lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php

```diff
@@ -192,7 +192,7 @@ public function getDropIndexSQL($index, $table = null)
 
         return "IF EXISTS (SELECT * FROM sysobjects WHERE name = '$index')
```

Comment:
@deeky666 you're right, but the sql generation for schema changes on SQL server has had a pretty rocky history. Schemas originally generated by doctrine are usually OK, but there are many issues when you build doctrine mappings for an existing schema and then attempt to generate a schema update.

As it stands, the SQL generated in the scenarios I outlined fails reliably, but shouldn't. So either the schema tool should be fixed, or the platform should be fixed to fail gracefully when it is provided garbage input.